ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/TransactionConsistencyValidator.scala

```diff
@@ -6,82 +6,74 @@ package com.daml.ledger.participant.state.kvutils.committer.transaction.validati
 import com.daml.ledger.participant.state.kvutils.Conversions
 import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
   DamlContractKey,
+  DamlContractKeyState,
+  DamlContractState,
   DamlStateKey,
-  DamlStateValue,
 }
-import com.daml.ledger.participant.state.kvutils.committer.transaction.validation.KeyMonotonicityValidation.checkContractKeysCausalMonotonicity
 import com.daml.ledger.participant.state.kvutils.committer.transaction.{
   DamlTransactionEntrySummary,
   Rejections,
   Step,
 }
 import com.daml.ledger.participant.state.kvutils.committer.{CommitContext, StepContinue, StepResult}
 import com.daml.ledger.participant.state.v1.RejectionReasonV0
-import com.daml.lf.data.Time.Timestamp
 import com.daml.lf.transaction.Transaction.{
   DuplicateKeys,
   InconsistentKeys,
   KeyActive,
   KeyCreate,
   NegativeKeyLookup,
 }
+import com.daml.lf.value.Value
 import com.daml.logging.LoggingContext
 
-private[transaction] object ContractKeysValidator extends TransactionValidator {
+private[transaction] object TransactionConsistencyValidator extends TransactionValidator {
 
-  /** Creates a committer step that validates casual monotonicity and consistency of contract keys
-    * against the current ledger state.
+  /** Validates consistency of contracts and contract keys against the current ledger state.
+    * For contracts, checks whether all contracts used in the transaction are still active.
```

Comment:
And for keys?

Reply:
If they are consistent and there are no duplicates. Happy to add something if there are any suggestions.

Reply:
```suggestion
    * For contracts, checks whether all contracts used in the transaction are still active.
    * For keys, checks whether they are consistent and there are no duplicates.
```

Reply:
Done